OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -1565,16 +1565,35 @@ internal static class DreamProcNativeRoot {
 
         [DreamProc("num2text")]
         [DreamProcParameter("N")]
-        [DreamProcParameter("Digits", Type = DreamValueTypeFlag.Float)]
-        [DreamProcParameter("Radix", Type = DreamValueTypeFlag.Float)]
+        [DreamProcParameter("A", Type = DreamValueTypeFlag.Float)]
+        [DreamProcParameter("B", Type = DreamValueTypeFlag.Float)]
         public static DreamValue NativeProc_num2text(NativeProc.Bundle bundle, DreamObject? src, DreamObject? usr) {
             DreamValue number = bundle.GetArgument(0, "N");
 
-            if (number.TryGetValueAsFloat(out float floatValue)) {
-                return new DreamValue(floatValue.ToString(CultureInfo.InvariantCulture));
-            } else {
+            if (!number.TryGetValueAsFloat(out float floatNum)) {
                 return new DreamValue("0");
             }
+            if(bundle.Arguments.Length == 1) {
+                return new DreamValue(floatNum.ToString("g6"));
+            }
+
+            if(bundle.Arguments.Length == 2) {
+                var sigFig = bundle.GetArgument(1, "A");
+                if(sigFig == DreamValue.Null) {
+                    return new DreamValue(floatNum.ToString("g6"));
+                }
+                return new DreamValue(floatNum.ToString('g' + sigFig.ToString()));
```

Comment:
This is not what `DreamValue.ToString()` is for, it should only be used to display a value to the user. This could very easily break at another time and does not work for non-integer values.
```suggestion
                if(!bundle.GetArgument(1, "A").TryGetValueAsInteger(out var sigFig)) {
                    return new DreamValue(floatNum.ToString("g6"));
                }
                return new DreamValue(floatNum.ToString($"g{sigFig}"));
```